### PR TITLE
fix: Include code in response error, fix tests

### PR DESCRIPTION
**Pull request by Filbird**
As part of the fix for https://github.com/screwdriver-cd/screwdriver/issues/450, in `getPermissions`, we want all scm-plugins to first check that a repository exists prior to fetching the permissions. This is implicit in `scm-gitlab` since the API for its [endpoint](https://docs.gitlab.com/ee/api/access_requests.html) will return a 404 if the repository does not exist.

In order to make this plugin compatible with the changes proposed in https://github.com/screwdriver-cd/screwdriver/pull/1187/files#diff-5a1574306e346e2ff33e2d567e18a944R60, we require that the `code` field of the response error is set.

Thanks!

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ describe('index', function () {
     });
 
     describe('parseUrl', () => {
-        const apiUrl = 'https://gitlab.com/api/v3/projects/batman%2Ftest';
+        const apiUrl = 'https://gitlab.com/api/v4/projects/batman%2Ftest';
```
Comment (bdangit) [inline test/index.test.js]: we're not actually hitting gitlab.com are we in our tests?

Comment (Filbird) [reply test/index.test.js]: I'm fairly certain we aren't. Seems like request is mocked out for all tests.

There are a few tests still failing however, may add an extra commit to fix those.

Comment (bdangit) [reply test/index.test.js]: Thanks @Filbird.

**Comment by bdangit** [comment]
👍

**Pull request merged**